jinja2/utils.py

```diff
@@ -284,6 +284,23 @@ def generate_lorem_ipsum(n=5, html=True, min=20, max=100):
     return Markup(u'\n'.join(u'<p>%s</p>' % escape(x) for x in result))
 
 
+def native_concat(invals):
+    '''
+    Jinja's native concat is just a list join. The list comes from result of
+    templating all the nodes. If the return type is expected to be
+    native, the list is artificial and we should return just the first
+    element.
+    '''
+    invals = [x for x in invals]
+    if isinstance(invals, list):
+        if len(invals) == 1:
+            # break out the single value
+            invals = invals[0]
+        elif len(invals) > 1:
+            # cast to unicode and join
+            invals = u''.join([u'%s' % x for x in invals])
```

Comment:
Why do you convert multiple nodes to a string? Wouldn't you want to return the list?